components/location/popup.jsx

```diff
@@ -0,0 +1,319 @@
+    >
+      {getTranslation("warm_up_here")}
+    </span>
+  );
+};
+
+// const Address = ({ address }) => {
+//   const googleMapDirections = `https://maps.google.com?saddr=Current+Location&daddr=${encodeURI(
+//     address
+//   )}`;
+//   return (
+//     <address>
+//       <a
+//         href={googleMapDirections}
+//         target="_blank"
+//         onClick="captureOutboundLink('${googleMapDirections}', 'directions')"
+//         rel="noreferrer"
+//       >
+//         {address}
+//       </a>
+//     </address>
+//   );
+// };
+
+const PublicTransit = ({ location }) => {
+  const options = location.publicTransitOptions;
+  const routes = options.map((option, i) => (
+    <React.Fragment key={`${option.routeName}-${i}`}>
+      {/** Kanad Note: I had to change all these span tags to div tags
+       * since it was causing issues when doing a ReactDOMServer render.
+       *
+       * I also got rid of the separate div with alt text and
+       * just added it as an aria-label below.
+       * Source: https://stackoverflow.com/a/38438002 */}
+      <div
+        aria-label={option.altText}
+        className="transit-route-option"
+        role="text"
+      >
+        <i className="material-icons-round route-icon">{option.icon}</i>
+        <h5
+          className="transit-route route-text"
+          style={{ backgroundColor: option.backgroundColor }}
+        >
+          {option.routeName}
+        </h5>
+        <h5 className="route-text">{option.distance}</h5>
+      </div>
+    </React.Fragment>
+  ));
+  return <PopupSection content={routes} id="public_transit" />;
+};
+
+// builds the section within the popup and replaces and URLs with links
+const SectionUrl = ({ id, value }) => {
+  const { getTranslation } = useContext(LanguageContext);
+  let urls = extractUrls(value);
+  let parsedText = parseLineBreaks(value);
+
+  if (urls) {
+    const distinctUrls = [...new Set(urls)];
+    distinctUrls.forEach((url) => {
+      let target_url = url;
+      if (!/[a-z]/i.test(url))
+        target_url = "tel:" + url.replace(/[^\d()-.]/g, "");
+      else if (/[\w.%+-]@[\w.]+/.test(url)) target_url = "mailto:" + url;
+      else if (!/http/i.test(url)) target_url = "http://" + url;
+
+      parsedText = parsedText.replace(
+        new RegExp(url, "g"),
+        // I removed on onClick handler here because nah
+        `<a href="${target_url}" target="_blank">${url}</a>`
+      );
+    });
+  }
+
+  return (
+    <div className="p row">
+      <p data-translation-id={id} className="key">
+        {getTranslation(id)}
+      </p>
+      <p
+        className="value"
+        dangerouslySetInnerHTML={{
+          __html: parsedText,
+        }}
+      />
+    </div>
+  );
+};
+
+const Popup = ({ getTranslation, language, location }) => (
+  <LanguageContext.Provider value={{ getTranslation, language }}>
+    {Object.keys(location).map((locationProperty, i) => {
+      switch (locationProperty) {
```

Comment:
This works as a one for one translation of what was there before, but I feel like it might make more sense to just build up the object the way it's supposed to be rather than looping through all of the properties each time...what do you think? If you want to hold off on that that's fine, but curious to hear your thoughts on the idea.

Reply:
Yeah I definitely agree... I tried to mimic the existing logic as much as possible but this is really difficult to follow and isn't all that efficient either. Will create a followup to fix this!